src/main/java/seam/SeamCalculator.java

```diff
@@ -31,116 +31,68 @@ public Pixel[][] calculateAndRemoveSeams()
         for (int verticalSeam = 0; verticalSeam < this.verticalSeams.length; verticalSeam++)
         {
             verticalSeams[verticalSeam] = calculateVerticalSeam(newPixels);
-            newPixels = removeVerticalSeam(verticalSeams[verticalSeam], newPixels);
+            Seam seam = verticalSeams[verticalSeam];
+            newPixels = recalculate(removeVerticalSeam(seam, newPixels));
```

Comment:
`recalcuate()` will get called an extra time at the last iteration of this for loop.

Reply:
I'm not sure I understand - what I think the code does is that it loops through the number of vertical seams and removes one by one modifying the newPixels[][] each time, and recalculating its value after every removal, and then do the same for all horizontal seams. Is that incorrect?

Reply:
Let's say you remove one seam.
After you remove that seam, it recalculates energies. That's a waste because you are done remove seams. So you have an extra calculation you don't need.